instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -185,6 +191,122 @@ def response_hook(span, instance, response):
 _INSTRUMENTATION_ATTR = "_is_instrumented_by_opentelemetry"
 
 
+def _get_redis_conn_info(instance):
+    host, port, db, unix_sock = None, None, None, None
+    pool = getattr(instance, "connection_pool", None)
+    if pool:
+        conn_kwargs = pool.connection_kwargs
+        host = conn_kwargs.get("host")
+        port = conn_kwargs.get("port")
+        db = conn_kwargs.get("db", 0)
+        unix_sock = conn_kwargs.get("path")
+    return host, port, db, unix_sock
+
+
+# Helper function to set old semantic convention attributes
+def _set_old_semconv_attributes(
+    span,
+    instance,
+    args,
+    query,
+    command_stack,
+    resource,
+    is_pipeline,
+):
+    span.set_attribute(DB_STATEMENT, query if not is_pipeline else resource)
+    _set_connection_attributes(span, instance)
+    if not is_pipeline:
+        span.set_attribute("db.redis.args_length", len(args))
+    else:
+        span.set_attribute("db.redis.pipeline_length", len(command_stack))
+
+
+# Helper function to set new semantic convention attributes
+def _set_new_semconv_attributes(
+    span,
+    instance,
+    args,
+    query,
+    command_stack,
+    resource,
+    is_pipeline,
+):
+    span.set_attribute("db.system.name", "redis")
+
+    if not is_pipeline:
+        if args and len(args) > 0:
+            span.set_attribute("db.operation.name", args[0])
+    else:
+        span.set_attribute("db.operation.name", "PIPELINE")
+        if len(command_stack) > 1:
+            span.set_attribute("db.operation.batch.size", len(command_stack))
+
+    host, port, db, unix_sock = _get_redis_conn_info(instance)
+    if db is not None:
+        span.set_attribute("db.namespace", str(db))
+    span.set_attribute("db.query.text", query if not is_pipeline else resource)
+    if host:
+        span.set_attribute("server.address", host)
+        span.set_attribute("network.peer.address", host)
+    if port:
+        span.set_attribute("server.port", port)
+        span.set_attribute("network.peer.port", port)
+    if unix_sock:
+        span.set_attribute("network.peer.address", unix_sock)
+        span.set_attribute("network.transport", "unix")
+
+    # db.stored_procedure.name (only for individual commands)
+    if not is_pipeline:
+        if args and args[0] in ("EVALSHA", "FCALL") and len(args) > 1:
+            span.set_attribute("db.stored_procedure.name", args[1])
+
+
+# Helper function to set all common span attributes
+def _set_span_attributes(
```

Comment:
There are some helpers in [util.py](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/34db73ebfd97761ba65523c3f07315ce2974dc78/instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/util.py) that look like they also collect values and set attributes. Would any of them help here, to reduce possible duplication?